model_interfaces/interface.py

```diff
@@ -41,7 +41,7 @@ def save_name(self):
         return f"{self.run_name} - {self.name}"
 
     @abstractmethod
-    def reply(self, user_message: str) -> str:
+    def reply(self, user_message: str, agent_response: str) -> str:
```

Comment:
It's a bit confusing that `agent_response` is mandatory. I'd maybe make it optional with a default value of `None`. Also don't forget that the empty string is also a valid message for an agent to give.

Reply:
We should figure this out. Reply is only called by `ChatSession.message_to_agent()` which has the optional `agent_response`, that that optional parameter is given to `reply`. Should the optional param in message_to_agent, also be optional for reply?.

Reply:
I don't think it needs to be optional, but the default value should be `None`. `None` means "ignore this value", and otherwise it should be taken as a response. But making it optional would feel better, sure.

Reply:
In my mind, all values that have defaults are optional. Since they are optional at call time.

Reply:
Ah right. Sorry. I expressed it badly. I meant that `None` should be given by default by `message_to_agent`, and then I think it's better if the parameter is optional in `reply`, because it's more flexible, but there's nothing forcing it to be right now.